plugins/woocommerce-admin/client/customize-store/assembler-hub/components/sidebar.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { createContext, useCallback, useState } from '@wordpress/element';
+import clsx from 'clsx';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export enum SidebarNavigationAnimationDirection {
+	Forward = 'forward',
+	Back = 'back',
+}
+
+type SidebarNavigationContextType = {
+	navigate: ( direction: SidebarNavigationAnimationDirection ) => void;
+};
+
+export const SidebarNavigationContext =
+	createContext< SidebarNavigationContextType >( {
+		navigate: () => void 0,
+	} );
+
+export function SidebarContent( { children }: { children: JSX.Element } ) {
+	const [ navState, setNavState ] = useState< {
+		direction: SidebarNavigationAnimationDirection | null;
+	} >( {
+		direction: null,
+	} );
+
+	const navigate = useCallback(
+		( direction: SidebarNavigationAnimationDirection ) => {
+			setNavState( { direction } );
+		},
+		[]
+	);
+
+	const wrapperCls = clsx( 'edit-site-sidebar__screen-wrapper', {
```

Comment:
Is the Customize Store UI really importing the `edit-site` stylesheet somewhere, and does it have the styles for the `edit-site-sidebar__content` and `edit-site-sidebar__screen-wrapper` classes defined?

Reply:
It's unfortunate that the Customize Store UI is using these `edit-site` CSS classes, as they are not a part of the `edit-site` public API and can change. I'm testing the Woo plugin on a plain WP 6.5.4 site, without the latest Gutenberg, and see two bugs:
- the slide-from-left and slide-from-right animations don't work
- when the browser window has low height, the sidebar should be scrollable, but it isn't. Here I can't scroll to "Choose your footer" and beyond:

<img width="799" alt="Screenshot 2024-06-10 at 18 50 55" src="https://github.com/woocommerce/woocommerce/assets/664258/7977acad-f8f4-4edd-a030-0e44c3ed612c">

The reason is that WP 6.5.4 doesn't yet have the `edit-site-sidebar__*` CSS changes from https://github.com/WordPress/gutenberg/pull/60849.

I think it's better to define your own classes. Or reuse something from WP Components if possible.

Reply:
Thnaks @jsnajdr! Good point! I created dedicated classes for this component with [90fba9e](https://github.com/woocommerce/woocommerce/pull/48312/commits/90fba9e51eec1483b4728eadc8a1e8c5f2b0c9d4).

Also, I created a dedicated issue to remove all the edit-site CSS classes from the CYS codebase! https://github.com/woocommerce/woocommerce/issues/48377